### PR TITLE
Dynamic Husk Icons

### PR DESCRIPTION
**Pull request by 13spacemen**
## About The Pull Request
Husk icons are now generated dynamically by greyscaling the limbs and applying the husk blood overlay
rather than everything using the same husk sprite
## Why It's Good For The Game
Less sprites needed, also good for nonhuman races especially those with non-humanoid shapes
All that is needed for a husk icon now is a blood overlay instead of an entire limb set

Human, Lizard, Moth, Monkey husks

PIC IS A BIT OUTDATED, THE BLOOD IN PIC IS DARKER THAN HUSK BLOOD (fixed) AND IT DOESNT "FIT" TO LIMB SPRITE (fixed)
Screenshot 2024-01-24 025229
## Changelog
:cl:
refactor: Husk icons are now dynamically generated. See if you can identify what species it was!
/:cl:

### CODE REVIEW

**Review by Wallemations — APPROVED** [review]
Idk about the code but sprites look fine

File: code/modules/surgery/bodyparts/_bodyparts.dm [hunk 1]
```diff
@@ -1066,6 +1062,18 @@
 
 	return .
 
+/obj/item/bodypart/proc/huskify_image(image/thing_to_husk, draw_blood = TRUE)
+	var/husk_color_mod = rgb(96, 88, 80)
```
Comment (Time-Green) [inline code/modules/surgery/bodyparts/_bodyparts.dm:1066]: Can you make a define for this instead? colors.dm has a list of similair usecases

File: code/modules/surgery/bodyparts/_bodyparts.dm [hunk 2]
```diff
@@ -1066,6 +1062,18 @@
...
+	var/icon/husk_icon = new(thing_to_husk.icon)
+	husk_icon.ColorTone(husk_color_mod)
+	thing_to_husk.icon = husk_icon
+	if(draw_blood)
+		var/mutable_appearance/husk_blood = mutable_appearance(icon_husk, "[husk_type]_husk_[body_zone]")
+		husk_blood.blend_mode |= BLEND_INSET_OVERLAY
+		husk_blood.appearance_flags |= RESET_COLOR
+		husk_blood.dir = thing_to_husk.dir
+		thing_to_husk.add_overlay(husk_blood)
```
Comment (Time-Green) [inline code/modules/surgery/bodyparts/_bodyparts.dm:1075]: I think this is okay? it's baked directly onto the limb code but we shouldn't get more like these

**Review by Time-Green — CHANGES_REQUESTED** [review]

**Comment by nikothedude** [comment]
a possibly unintended balance change of this PR is somewhat nerfing the identity erasure aspect of husks

as it stands husks hide all external organs and apply the same sprite to everything as well as obscuring the name of the husked person, making them completely unidentifiable without extracting like, blood

File: code/modules/surgery/bodyparts/_bodyparts.dm (hunk 2 above)
Comment (Time-Green) [reply code/modules/surgery/bodyparts/_bodyparts.dm:1075]: Does curing clear the husks overlay? Or do they stack infinitely? You're technically adding overlays outside of the usual path

**Comment by 13spacemen** [comment]
@Time-Green added the define
also tested with become_husk() and cure_husk() proccalls, and dismemberment
seems to work fine no duplicate overlays

File: code/modules/surgery/bodyparts/_bodyparts.dm (hunk 2 above)
Comment (13spacemen) [reply code/modules/surgery/bodyparts/_bodyparts.dm:1074]: clears the overlay

**Pull request merged**